panther_lights_controller/src/panther_apa102_driver.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+from threading import Lock
+from typing import Optional
+
+from apa102_pi.driver import apa102
+import RPi.GPIO as GPIO
+
+
+class PantherAPA102Driver:
+    LED_SWITCH_FRONT_STATE = GPIO.HIGH
+    LED_SWITCH_REAR_STATE = GPIO.LOW
+    LED_POWER_ON_STATE = GPIO.LOW  # active LED with low state
+
+    def __init__(
+        self,
+        num_led,
+        panel_count: Optional[int] = 2,
+        brightness: Optional[int] = 31,
+        led_switch_pin: Optional[int] = 20,
+        led_power_pin: Optional[int] = 26,
+    ) -> None:
+
+        self._num_led = num_led
+        self._panel_count = panel_count
+
+        self._is_running = True
+        self._thread_finished = False
+        self._lock = Lock()
+        self._pixels = apa102.APA102(
+            num_led=self._num_led, order='rgb', mosi=10, sclk=11, global_brightness=brightness
+        )
+        self._led_switch_pin = led_switch_pin
+        self._led_power_pin = led_power_pin
+        self._front_active = True
+        self._rear_active = True
+
+        # setup and activate panels
+        self._setup_panels()
+
+    def _setup_panels(self) -> None:
+        GPIO.setmode(GPIO.BCM)
+        GPIO.setup(self._led_switch_pin, GPIO.OUT)
+        GPIO.setup(self._led_power_pin, GPIO.OUT)
+        GPIO.output(
+            (self._led_switch_pin, self._led_power_pin),
+            (
+                PantherAPA102Driver.LED_SWITCH_FRONT_STATE,
+                PantherAPA102Driver.LED_POWER_ON_STATE,
+            ),
+        )
+
+    def set_panel_frame(self, panel_num, panel_frame, brightness: Optional[int] = 100) -> None:
+        with self._lock:
+            # select panel
+            if panel_num == 0 and self._front_active:
+                GPIO.output(self._led_switch_pin, PantherAPA102Driver.LED_SWITCH_FRONT_STATE)
+            elif panel_num == 1 and self._rear_active:
+                GPIO.output(self._led_switch_pin, PantherAPA102Driver.LED_SWITCH_REAR_STATE)
+            else:
+                raise ValueError('panther lights have only two panels')
+
+            # set all LED in this panel
+            for i in range(self._num_led):
+                self._pixels.set_pixel_rgb(i, int(panel_frame[i]), brightness)
+            self._pixels.show()
+
+    def set_panel_state(self, panel_num, state) -> None:
```

Comment:
```suggestion
    def set_panel_state(self, panel_num: int, state: bool) -> None:
```
I hope it is a good type of `state` variable